### ISSUE TITLE
Wrong VulkanHandle Equals

### ISSUE THREAD

**Issue by amerkoleci** [discosultan/VulkanCore#1]
The operator == is wrong of VulkanHandle and does not compare correct when left is null:

Should be

```
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static bool operator ==(VulkanHandle<THandle> left, VulkanHandle<THandle> right)
{
	if (ReferenceEquals(left, right))
	{
		return true;
	}

	return left?.Equals(right) ?? false;
}
```

**Comment by discosultan** [comment]
Hey! Sorry for late reply, but I only noticed this issue now ^^. It must have slipped past me somehow.

Could you provide a simple failing sample?

-------

Currently the `==` operator is overloaded as follows:

```csharp
public static bool operator ==(VulkanHandle<THandle> left, VulkanHandle<THandle> right)
{
    left?.Equals(right) ?? false;
}
```
If only `left` is `null`, they are not considered equal. If both `left` and `right` are null, they are also not considered equal (similar to how `null` compares to `null` by default).

This `==` operator in turn calls out the overloaded `Equals` method which is defined as follows:

```csharp
public override bool Equals(object obj)
{
    if (ReferenceEquals(null, obj)) return false;
    if (ReferenceEquals(this, obj)) return true;
    if (obj.GetType() != GetType()) return false;
    return Handle.Equals(((VulkanHandle<THandle>)obj).Handle);
}
```

In which the reference equality is included.

------

On a side note, I was hesitant to overload equality for *handle* types in the first place and I might remove it at some point (unless proven that it's of any use).

**Comment by amerkoleci** [comment]
Of course, take look at this code:

```
private static Instance _instance;

pubic void SomeMethod() 
{
           if(_instance == null)  // This does not work
              _instance = new Instance();
}

```

**Comment by discosultan** [comment]
Aha, of course - thanks for pointing this out!

I've decided to remove the entire custom equality comparison for handles. It's sole purpose was to provide a shorthand to compare if two instances of `VulkanHandle{}` classes contain the same underlying handle, but it might as well be done by comparing `VulkanHandle{}.Handle` properties directly.

I'll push a new nuget version soon.

**Comment by discosultan** [comment]
New `alpha11` nuget package is now available.

------

I've removed custom equality comparison for all handle types (https://github.com/discosultan/VulkanCore/commit/0ecca37c3525ca2e316cd0e8df8f56c7caa9e374) to reduce complexity/confusion. It didn't really provide any value.

-----

Since there are no release notes available for `alphaXX` releases, I'll also mention here that I've also removed `Image.BackingMemory` and `Buffer.BackingMemory` properties (https://github.com/discosultan/VulkanCore/commit/d4229876e02d00eddb0dc4fa35ec658c768f823c). The reasoning here being that the property did not properly reflect bound memory in cases where an image was acquired from a swapchain, for example. It should be application's responsibility to track buffer and image resources along with their bound memory handles.
